src/Microsoft.ML.TimeSeries/IidAnomalyDetectionBase.cs

```diff
@@ -13,99 +13,147 @@
 namespace Microsoft.ML.TimeSeriesProcessing
 {
     /// <summary>
-    /// This transform computes the p-values and martingale scores for a supposedly i.i.d input sequence of floats. In other words, it assumes
+    /// The wrapper to transform that computes the p-values and martingale scores for a supposedly i.i.d input sequence of floats. In other words, it assumes
     /// the input sequence represents the raw anomaly score which might have been computed via another process.
     /// </summary>
-    public abstract class IidAnomalyDetectionBase : SequentialAnomalyDetectionTransformBase<Single, IidAnomalyDetectionBase.State>
+    public class IidAnomalyDetectionBaseWrapper : IStatefulTransformer, ICanSaveModel
     {
-        public IidAnomalyDetectionBase(ArgumentsBase args, string name, IHostEnvironment env)
-            : base(args, name, env)
-        {
-            InitialWindowSize = 0;
-            StateRef = new State();
-            StateRef.InitState(WindowSize, InitialWindowSize, this, Host);
-        }
+        public bool IsRowToRowMapper => Base.IsRowToRowMapper;
 
-        public IidAnomalyDetectionBase(IHostEnvironment env, ModelLoadContext ctx, string name)
-            : base(env, ctx, name)
-        {
-            Host.CheckDecode(InitialWindowSize == 0);
-            StateRef = new State(ctx.Reader);
-            StateRef.InitState(this, Host);
-        }
+        IStatefulTransformer IStatefulTransformer.Clone() => Base.Clone();
+
+        public Schema GetOutputSchema(Schema inputSchema) => Base.GetOutputSchema(inputSchema);
+
+        public IRowToRowMapper GetRowToRowMapper(Schema inputSchema) => Base.GetRowToRowMapper(inputSchema);
```

Comment:
Do we want to expose this function to users? If not, maybe consider explicit interface implementation?

Reply:
Same here, I agree that it is a bit confusing, I might consider doing what I am doing with ICanSaveModel, and explicit implement the interface for these methods, but it should be done in another PR, and further discussed.

---
In reply to: [254477452](https://github.com/dotnet/machinelearning/pull/2344#discussion_r254477452) [](ancestors = 254477452)

Reply:
Agreed, this needs to be its own change since it require creating and implementing a new interface.

---
In reply to: [254485970](https://github.com/dotnet/machinelearning/pull/2344#discussion_r254485970) [](ancestors = 254485970,254477452)

Reply:
Could you please open an issue?

---
In reply to: [255151564](https://github.com/dotnet/machinelearning/pull/2344#discussion_r255151564) [](ancestors = 255151564,254485970,254477452)